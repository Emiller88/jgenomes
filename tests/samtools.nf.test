nextflow_pipeline {

    name "Build samtools references"
    script "../main.nf"
    tag "pipeline"

    test("Run with profile test for samtools | --input GRCh38_chr21.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = 'https://raw.githubusercontent.com/nf-core/test-datasets/refs/heads/references/references/GRCh38_chr21/GRCh38_chr21.yml'
                outdir = "$outputDir"
                tools  = 'faidx,intervals,sizes'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("Run with profile test for samtools | --input assets/test/default_extended.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = "${projectDir}/assets/test/default_extended.yml"
                outdir = "$outputDir"
                tools  = 'faidx,intervals,sizes'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

    test("Run with profile test for samtools | --input assets/test/default_full.yml") {

        options "-output-dir $outputDir"

        when {
            params {
                input  = "${projectDir}/assets/test/default_full.yml"
                outdir = "$outputDir"
                tools  = 'faidx,intervals,sizes'
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_references_software_mqc_versions.yml"),
                    // All stable path name, with a relative path
                    stable_name,
                    // All files with stable contents
                    stable_path
                ).match() }
            )
        }
    }

}
