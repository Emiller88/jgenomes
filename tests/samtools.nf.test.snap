{
    "Run with profile test for samtools | --input assets/test/default_full.yml": {
        "content": [
            1,
            {
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "index.json",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml"
            ],
            [
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-28T13:03:25.374670864"
    },
    "Run with profile test for samtools | --input GRCh38_chr21.yml": {
        "content": [
            3,
            {
                "BUILD_INTERVALS": {
                    "gawk": "5.3.0"
                },
                "SAMTOOLS_FAIDX": {
                    "samtools": 1.21
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "Homo_sapiens",
                "Homo_sapiens/nf-core",
                "Homo_sapiens/nf-core/references",
                "Homo_sapiens/nf-core/references/GRCh38_chr21",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation/intervals",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation/intervals/GRCh38_chr21.bed",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/WholeGenomeFasta",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/WholeGenomeFasta/GRCh38_chr21.fa.fai",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Sequence/WholeGenomeFasta/GRCh38_chr21.fa.sizes",
                "index.json",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml"
            ],
            [
                "GRCh38_chr21.bed:md5,472d213cfcde96565699779d5bfc0e32",
                "GRCh38_chr21.fa.fai:md5,6ea51e1c1231da5e49212ea9f493d703",
                "GRCh38_chr21.fa.sizes:md5,c985a141685e8431ec27c782816cb744",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-03T13:07:34.895283019"
    },
    "Run with profile test for samtools | --input assets/test/default_extended.yml": {
        "content": [
            2,
            {
                "BUILD_INTERVALS": {
                    "gawk": "5.3.0"
                },
                "Workflow": {
                    "nf-core/references": "v1.0dev"
                }
            },
            [
                "Homo_sapiens",
                "Homo_sapiens/nf-core",
                "Homo_sapiens/nf-core/references",
                "Homo_sapiens/nf-core/references/GRCh38_chr21",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation/intervals",
                "Homo_sapiens/nf-core/references/GRCh38_chr21/Annotation/intervals/GRCh38_chr21.bed",
                "index.json",
                "multiqc",
                "multiqc/multiqc_data",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_report.html",
                "pipeline_info",
                "pipeline_info/nf_core_references_software_mqc_versions.yml"
            ],
            [
                "GRCh38_chr21.bed:md5,472d213cfcde96565699779d5bfc0e32",
                "multiqc_citations.txt:md5,4c806e63a283ec1b7e78cdae3a923d4f"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-11-28T14:47:38.70092941"
    }
}
